### PR TITLE
Add new orb to handle versioning of packages and services

### PR DESCRIPTION
**Pull request by YBadiss**

### CODE REVIEW

File: src/versioning/orb.yml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    steps:
+      - checkout
+      - set_version_command
+      - set_new_commit_and_version_env_vars
+      - run:
+          name: Push new dist to gemfury
+          command: |
+            if [[ "${IS_PYTHON_PACKAGE}" == "1" ]]; then
+              python setup.py sdist
+              curl --fail -F package=@dist/python-ledgercommon-${NEW_VERSION}.tar.gz https://${PYPI_PUSH_TOKEN}@push.fury.io/ledger/
```
Comment (CedricCabessa) [inline src/versioning/orb.yml:96]: not ledgercommon !

File: src/versioning/orb.yml [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      - set_version_command
+      - set_new_commit_and_version_env_vars
+      - set_old_commit_and_version_env_vars
+      - run:
+          name: Install python-packaging
+          command: pip install -U --user packaging
+      - run:
+          name: Check version
+          command: |
+            python -c "from packaging import version; assert ('${NEW_COMMIT}' == '${OLD_COMMIT}') or (version.parse('${NEW_VERSION}') > version.parse('${OLD_VERSION}'))"
```
Comment (CedricCabessa) [inline src/versioning/orb.yml:82]: I don't understand in what situation we get NEW_COMMIT == OLD_COMMIT ? :thinking:

File: src/versioning/orb.yml (hunk 1 above)
Comment (YBadiss) [reply src/versioning/orb.yml:96]: ah yes thanks 😅

File: src/versioning/orb.yml (hunk 2 above)
Comment (YBadiss) [reply src/versioning/orb.yml:82]: after a merge, circleci builds the target merged branch (develop, master, ...). I renamed `OLD_COMMIT` to `MASTER_COMMIT` as that's what it is, and this case is post merge of the PR.

Comment (YBadiss) [reply src/versioning/orb.yml:82]: actually I simplified this a lot, and added a comment explaining why

**Pull request merged**